packages/labs/task/src/task.ts

```diff
@@ -99,14 +96,14 @@ export interface TaskConfig<T extends unknown[], R> {
  * }
  */
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-export class Task<T extends [...unknown[]] = any, R = any> {
+export class Task<T extends unknown[] = any, R = any, E = any> {
```

Comment:
Since you're casting all instances of `E` to `as E`, do you even need `E` on `Task`? can't `_error` simply just be `Error` and `StatusRenderer<R, E>` be `StatusRenderer<R, E extends Error = Error>`?